getting_started_modules.md

```diff
@@ -0,0 +1,341 @@
+If we are provisioning a postgres service, maybe we'd like a "Name of postgres database" variable as input, or "Number of servers to provision" if you are provisioning a cluster.  
+A variable is defined like below:
+
+```hcl-terraform
+variable "service_name" {
+  type        = string
+  description = "Minio service name"
+  default     = "minio"
+}
+```
+
+##### outputs.tf
+
+> :warning: In the first iteration we don't need any output variables.
+
+This files contains variables that will be available as [output variables](https://www.terraform.io/docs/configuration/outputs.html) when you use a module. 
+Below is first an example of how to define output-variables, then an example of how to use a module, and access their output variables.
+Defining output variables:
+
+ ```hcl-terraform
+output "nomad_job" {
+  value       = nomad_job.countdash
+  description = "The countdash nomad job object"
+}
+```
+
+> :bulb: Together inputs and outputs should create a very clear picture of how a module should be used. 
+>For example in our hive module we have clearly defined that it needs to have a postgres-address as an input. 
+>In our postgres module we have an output that is exactly that. In other words, we might need to import and setup a postgres-module before setting up our hive-module, so that we get a postgres-address to give our hive-module. 
+>Or, if we already have a postgres-address available, we could supply that instead. The goal is to clearly define the needs of a module, while at the same time making it flexible and generic (in the example of hive we give the user the ability to use any postgres they'd like). 
+>How to use variables and outputs will be shown later in [make the module more dynamic]().
+
+
+#### 4. Using a Terraform Module
+At this point we have created three files (or one): 
+
+- `main.tf` 
+- `variables.tf` 
+- `outputs.tf` 
+
+Together they do one thing: Start a nomad job. 
+Aren't we done with our goal then? Almost. We still need to verify that it works the way we expect it to. 
+Next step is to write some terraform code that runs the module we just made. Then we'll import that code to our virtual machine and run it there. 
+There are two steps to this:
+1. Write the code that uses the module, and run it manually inside the virtual machine 
+2. Write ansible code that'll run it automatically when you start the box.
+
+Below is an example on how to use a module:
+
+ ```hcl-terraform
+module "minio" {
+  source      = "github.com/fredrikhgrelland/terraform-nomad-minio.git?ref=0.0.3"
+}
+```
+
+This will fetch the module at the given source. 
+In the case above it is a MinIO module, version 0.0.3. 
+
+
+##### a. How To Use The Module and Run It Manually
+Create a file called `main.
+tf` under the `example/` directory, and add the code above, change the module's name, and source. The source is `../` 
+in this case because that is where our module files are, relative to the `main.tf` you are writing this in. 
```

Comment:
```suggestion
Create a file called `main.tf` under the `example/` directory, and add the code above, change the module's name, and source.
The source is `../` in this case because that is where our module files are, relative to the `main.tf` you are writing this in. 
```

Reply:
already fixed this, resolving